db/helpers.go

```diff
@@ -129,6 +129,13 @@ func (ms *MongoStorage) createIndexes() error {
 	if _, err := ms.verifications.Indexes().CreateOne(ctx, verificationCodeIndex); err != nil {
```

Comment:
@lucasmenendez a late comment that I should have caught in a previous PR.
The index is there to help the search. IMHO, since in your searches you use the tuple `(code,type)` then you should either update your index or the search filter.